megamek/src/megamek/utils/MegaMekXmlUtil.java

```diff
@@ -129,94 +139,321 @@ public static XMLReader createSafeXMLReader() {
+            pw.println(indentStr(indent) + "<" + name + ">" + stringJoiner + "</" + name + ">");
         }
     }
 
-    public static void writeSimpleXmlTag(PrintWriter pw1, int indent, String name, int val) {
-        pw1.println(indentStr(indent) + "<" + name + ">" + val + "</" + name + ">");
+    /**
+     * This writes a collection of objects that are saved based on their toString value.
+     * Examples are UUID and LocalDate
+     * @param pw the PrintWriter to use
+     * @param indent the indent to write at
+     * @param name the name of the XML tag
+     * @param values the collection to write to XML
+     */
+    public static void writeSimpleXMLTag(final PrintWriter pw, final int indent, final String name,
+                                         final Collection<?> values) {
+        if (values.size() > 0) {
+            final StringJoiner stringJoiner = new StringJoiner(",", "", "");
+            values.forEach(v -> stringJoiner.add(v.toString()));
+            pw.println(indentStr(indent) + "<" + name + ">" + stringJoiner + "</" + name + ">");
+        }
     }
 
-    public static void writeSimpleXMLTag(PrintWriter pw1, int indent, String name, int... values) {
+    /**
+     * This writes a String or an array of Strings to file
+     * @param pw the PrintWriter to use
+     * @param indent the indent to write at
+     * @param name the name of the XML tag
+     * @param values the String or String[] to write to XML
+     */
+    public static void writeSimpleXMLTag(final PrintWriter pw, final int indent, final String name,
+                                         final String... values) {
+        writeSimpleXMLAttributedTag(pw, indent, name, null, null, values);
+    }
+
+    /**
+     * This writes a String or an array of Strings to file, with an the possible addition of an
+     * attribute and its value
+     * @param pw the PrintWriter to use
+     * @param indent the indent to write at
+     * @param name the name of the XML tag
+     * @param attributeName the attribute to write as part of the XML tag
+     * @param attributeValue the value of the attribute
+     * @param values the String or String[] to write to XML
+     */
+    public static void writeSimpleXMLAttributedTag(final PrintWriter pw, final int indent,
+                                                   final String name,
+                                                   final @Nullable String attributeName,
+                                                   final @Nullable String attributeValue,
+                                                   final String... values) {
         if (values.length > 0) {
-            pw1.println(indentStr(indent) + "<" + name + ">" + StringUtils.join(values, ',') + "</" + name + ">");
+            final boolean hasAttribute = attributeValue != null;
+            pw.print(indentStr(indent) + "<" + name);
+            if (hasAttribute) {
+                pw.print(" " + attributeName + "=\"" + attributeValue + "\"");
```

Comment:
`attributeValue` should likely be escaped too.

Reply:
It should also be T, not String